torchtext/datasets/cnndm.py

```diff
@@ -0,0 +1,100 @@
+import os
+from functools import partial
+from typing import Union, Tuple
+
+from torchtext._internal.module_utils import is_module_available
+from torchtext.data.datasets_utils import (
+    _wrap_split_argument,
+    _create_dataset_directory,
+)
+
+if is_module_available("torchdata"):
+    from torchdata.datapipes.iter import (
+        FileOpener,
+        IterableWrapper,
+        OnlineReader,
+        GDriveReader,
+    )
+
+DATASET_NAME = "CNNDM"
+
+URL_LIST = {
+    "train": "https://raw.githubusercontent.com/abisee/cnn-dailymail/master/url_lists/all_train.txt",
+    "val": "https://raw.githubusercontent.com/abisee/cnn-dailymail/master/url_lists/all_val.txt",
+    "test": "https://raw.githubusercontent.com/abisee/cnn-dailymail/master/url_lists/all_test.txt",
+}
+
+URL_LIST_MD5 = {
+    "train": "c8ca98cfcb6cf3f99a404552568490bc",
+    "val": "83a3c483b3ed38b1392285bed668bfee",
+    "test": "4f3ac04669934dbc746b7061e68a0258",
+}
+
+STORIES_LIST = {
+    "cnn": "https://drive.google.com/uc?export=download&id=0BwmD_VLjROrfTHk4NFg2SndKcjQ",
+    "dailymail": "https://drive.google.com/uc?export=download&id=0BwmD_VLjROrfM1BxdkxVaTY2bWs",
+}
+
+PATH_LIST = {
+    "cnn": "cnn_stories.tgz",
+    "dailymail": "dailymail_stories.tgz",
+}
+
+STORIES_MD5 = {"cnn": "85ac23a1926a831e8f46a6b8eaf57263", "dailymail": "f9c5f565e8abe86c38bfa4ae8f96fd72"}
+
+_EXTRACTED_FOLDERS = {
+    "cnn": os.path.join("cnn", "stories"),
+    "daily_mail": os.path.join("dailymail", "stories"),
+}
+
+
+def _filepath_fn(root: str, source: str, _=None):
+    return os.path.join(root, PATH_LIST[source])
+
+
+def _extracted_filepath_fn(root: str, source: str, t):
```

Comment:
leave a comment here saying that this function will be used to cache the contents of the tar file